Objects/setobject.c

```diff
@@ -1095,6 +1095,41 @@ frozenset_new(PyTypeObject *type, PyObject *args, PyObject *kwds)
     return emptyfrozenset;
 }
 
+static PyObject *
+frozenset_vectorcall(PyObject *type, PyObject * const*args,
+                     size_t nargsf, PyObject *kwnames)
+{
+    assert(PyType_Check(type));
+
+    if (!_PyArg_NoKwnames("frozenset", kwnames)) {
+        return NULL;
+    }
+
+    Py_ssize_t nargs = PyVectorcall_NARGS(nargsf);
+    if (!_PyArg_CheckPositional("frozenset", nargs, 0, 1)) {
+        return NULL;
+    }
+
+    if (nargs) {
```

Comment:
Can't we share more code with frozenset_new()? It looks like most code is duplicated. Add an private helper function and move the code which is common there.

Reply:
> Can't we share more code with frozenset_new()?

Good idea :)